File: GestionCommande/src/Model/Commande.ts
import mongoose from "mongoose";
import mongoosePaginate from "mongoose-paginate"

let CommandeSchema=new mongoose.Schema({
    NumCommande:{
        type:String,
        required:true,
        unique:true
    },
    Name:{
        type:String,
        required:true
    },
    Cin:{
        type:String,
        required:true
    },
    LastName:{
        type:String,
        required:true
    },
    email:{
        type:String,
        required:true
    },
    phone:{
        type:String,
        required:true
    },
    PrixTotal:{
        type:String,
        default: null
    },
    location: {
		latitude: {
             type: String,
             required: true 
        },
		longitude: { 
            type: String,
            required: true
         },
         name:{
            type: String,
            required: true 
         }
	},
    Client_id:{
        type:String,
        required:true
    },
    Status:{
        type:String,
        required:true,
        default:"Waiting"
    },
    Date:{
        type:Date,
        default:Date.now
    },
    LigneCommandes: [{ type: mongoose.Schema.Types.ObjectId, ref: 'LigneCommande' }]
})

CommandeSchema.plugin(mongoosePaginate)
const Commande=mongoose.model("Commande",CommandeSchema)
export default Commande;

File: deliverymarket/lib/Models/Commande.dart
class Commande {
  String NumCommande;
  String Name;
  String Cin;
  String LastName;
  String email;
  String phone;
  String PrixTotal; 
  Map<String, dynamic> location; 
  String Client_id;
  String Status;
  DateTime Date;
  List<Map<String, dynamic>> LigneCommandes; 

  Commande({
    required this.NumCommande,
    required this.Name,
    required this.Cin,
    required this.LastName,
    required this.email,
    required this.phone,
    required this.LigneCommandes,
    required this.Client_id,
    required this.Status,
    required this.location,
    required this.PrixTotal,
    required this.Date,
  });

  factory Commande.fromJson(Map<String, dynamic> json) {
    return Commande(
      NumCommande: json['NumCommande'],
      Name: json['Name'],
      Cin: json['Cin'],
      LastName: json['LastName'],
      email: json['email'],
      phone: json['phone'],
      LigneCommandes: List<Map<String, dynamic>>.from(json['LigneCommandes']),
      Client_id: json['Client_id'],
      Status: json['Status'],
      location: Map<String, dynamic>.from(json['location']), 
      PrixTotal: json['PrixTotal'], 
      Date: DateTime.parse(json['Date']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'NumCommande': NumCommande,
      'Name': Name,
      'Cin': Cin,
      'LastName': LastName,
      'email': email,
      'phone': phone,
      'LigneCommandes': LigneCommandes,
      'Client_id': Client_id,
      'Status': Status,
      'location': location,
      'PrixTotal': PrixTotal,
      'Date': Date.toIso8601String(),
    };
  }

  @override
  String toString() {
    return 'Commande { LigneCommandes:$LigneCommandes NumCommande:$NumCommande,Name:$Name ,Cin:$Cin,LastName:$LastName,email:$email,phone:$phone  Client_id:$Client_id Status:$Status location:$location PrixTotal:$PrixTotal Date:$Date}';
  }
}
